### PR TITLE
falcon refactoring with response writer interception support

### PR DESCRIPTION
**Pull request by maxkaneco**

### CODE REVIEW

File: node/handlers/handlers.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package handlers
```
Comment (maxkaneco) [inline node/handlers/handlers.go:1]: no code change

File: node/handlers/name.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package handlers
```
Comment (maxkaneco) [inline node/handlers/name.go:1]: no code change

File: node/handlers/pin.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package handlers
```
Comment (maxkaneco) [inline node/handlers/pin.go:1]: no code change

File: node/handlers/status.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package handlers
```
Comment (maxkaneco) [inline node/handlers/status.go:1]: no code change

File: node/metrics.go [hunk 5]
```diff
@@ -29,13 +33,35 @@ func initMetrics(ctx context.Context, port int) {
...
-	metrics.NewCounter("rc_pinner_unpin_err_total")
-	metrics.NewCounter("rc_pinner_recursive_keys_call_total")
+	com.RegisterPinnerMetrics()
 	metrics.NewGauge("pinned_count_total")
 	metrics.NewGauge("connected_peers_total")
 
 	metrics.GaugeSet("restart_at_seconds", float64(time.Now().Unix()))
 }
+
+func updateNodeMetrics(
```
Comment (maxkaneco) [inline node/metrics.go:43]: no code change

**Review by jzhu-eth — APPROVED** [review]

**Pull request merged**